File: 605.can-place-flowers.py
#
# @lc app=leetcode id=605 lang=python3
#
# [605] Can Place Flowers
#

# @lc code=start
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        cleaned_flowerbed = [0] + flowerbed + [0] 

        allowed_plot_num = 0
        for i in range(1, len(cleaned_flowerbed) - 1):
            if cleaned_flowerbed[i - 1] == cleaned_flowerbed[i] == cleaned_flowerbed[i + 1] == 0:
                cleaned_flowerbed[i] = 1
                allowed_plot_num += 1
        
        return allowed_plot_num >= n
        
        
# @lc code=end

File: 605.can-place-flowers.swift
/*
 * @lc app=leetcode id=605 lang=swift
 *
 * [605] Can Place Flowers
 */

// @lc code=start
class Solution {
    func canPlaceFlowers(_ flowerbed: [Int], _ n: Int) -> Bool {
        var new_flowerbed = [0] + flowerbed + [0]
        var count = 0

        for i in 1..<new_flowerbed.count - 1 {
            if new_flowerbed[i - 1] == 0 && new_flowerbed[i] == 0 && new_flowerbed[i + 1] == 0 {
                new_flowerbed[i] = 1
                count += 1
            }
        }
        return count >= n
    }
}
// @lc code=end

